keps/core/model-cross-worksapce-entitlements.md

```diff
@@ -0,0 +1,1004 @@
+
+    // Children determines if this policy binding provides the entitlement just to the
+    // workspace in which it is created, or also to the child workspaces underneath that.
+    // Optional, defaults to false.
+    // +optional
+    Children boolean `json:"children,omitempty"`
+}
+
+type EntitlementPolicyRef struct {
+    // ClusterPath is the human-readable path to the logical cluster in which the entitlement
+    // policy is registered.
+    ClusterPath logicalcluster.Path `json:"clusterPath"`
+
+    // Name is the name of the entitlement policy.
+    Name string `json:"name"`
+}
+```
+
+A service provider with permissions to create `EntitlementPolicy` will do so using the
+normal `kcp` API:
+
+```
+> POST /clusters/<entitlement-provider-cluster>/apis/entitlements.tenancy.kcp.io/v1alpha1/entitlementpolicies
+> {
+>     "kind": "EntitlementPolicy",
+>     "apiVersion": "entitlements.tenancy.kcp.io/v1alpha1",
+>     "metadata": {
+>         "name": "something-meaningful"
+>     },
+>     "entitlements": [
+>         {
+>             "clusterName": "33bab531",
+>             "kind": "WidgetAccess",
+>             "apiVersion": "acme.corp/v1",
+>             "spec": {
+>                 "something": {
+>                     "meaningful": "here"
+>                 }
+>             }
+>         }
+>     ]
+> }
+201 CREATED
+{
+    "kind": "EntitlementPolicy",
+    "apiVersion": "entitlements.tenancy.kcp.io/v1alpha1",
+    "metadata": {
+        "name": "something-meaningful"
+    },
+    "entitlements": [
+        {
+            "clusterName": "33bab531",
+            "kind": "WidgetAccess",
+            "apiVersion": "acme.corp/v1",
+            "spec": {
+                "something": {
+                    "meaningful": "here"
+                }
+            }
+        }
+    ]
+}
+```
+
+However, to entitle tenants, a per-tenant endpoint is used:
+
+```
+> POST /services/entitlements/clusters/<entitlement-consumer-cluster>/apis/entitlements.tenancy.kcp.io/v1alpha1/entitlementpolicybindings
+> {
+>     "kind": "EntitlementPolicyBinding",
+>     "apiVersion": "entitlements.tenancy.kcp.io/v1alpha1",
+>     "metadata": {
+>         "name": "something-meaningful-for-a-tenant"
+>     },
+>     "entitlementPolicyRef": {
+>         "clusterPath": "<entitlement-provider-cluster>",
+>         "name": "something-meaningful"
+>     },
+>     children: false
+> }
+201 CREATED
+{
+    "kind": "EntitlementPolicyBinding",
+    "apiVersion": "entitlements.tenancy.kcp.io/v1alpha1",
+    "metadata": {
+        "name": "something-meaningful-for-a-tenant"
+    },
+    "entitlementPolicyRef": {
+        "clusterPath": "<entitlement-provider-cluster>",
+        "name": "something-meaningful"
+    },
+    children: false
+}
+```
+
+Entitlement policy bindings will be stored in tenant clusters; however, the tenant will have
+no access to read or change them. This mechanism is used to distribute this data, reduce its
```

Comment:
why not read?

Reply:
I was hoping to strike a middle ground between providing a useful base implementation while not exposing the APIs anywhere they don't need to be exposed, so it's less likely that when a deployment of `kcp` changes the implementation mechanism, people have their workflows broken. Perhaps useful to sketch out what it looks like when a custom entitlement review is added, though - no real reason to remove this one from the system at that time, it could still function, as a fallback?

Reply:
I like that thought :+1: If you cannot read them, nobody will ask about listing either, and with that nobody will ask for an exhaustive list of entitlement including external ones.